### PR TITLE
enterprise optimizations

### PR DESCRIPTION
**Pull request by snopoke**
@dimagi/scale-team 

I included a few tweaks to toggle logic as well, mainly that `enabled_for_request` wasn't respecting the hard coded items.

### CODE REVIEW

File: corehq/apps/domain_migration_flags/api.py [hunk 1]
```diff
@@ -41,7 +41,7 @@ def get_migration_complete(domain, slug):
     return get_migration_status(domain, slug) == MigrationStatus.COMPLETE
 
 
-@quickcache(['domain', 'slug'], skip_arg='strict')
+@quickcache(['domain', 'slug'], skip_arg='strict', timeout=60 * 60)
```
Comment (dannyroberts) [inline corehq/apps/domain_migration_flags/api.py]: Is it also worth setting the `memoized_timeout` to longer (defaults to 10 seconds)?

File: corehq/toggles.py [hunk 2]
```diff
@@ -68,22 +68,24 @@ def __init__(self, slug, label, tag, namespaces=None, help_link=None,
...
 
-    def enabled(self, item, **kwargs):
+    def enabled(self, item, namespace=Ellipsis):
         if item in self.always_enabled:
             return True
         elif item in self.always_disabled:
             return False
-        return any([toggle_enabled(self.slug, item, namespace=n, **kwargs) for n in self.namespaces])
+
+        namespaces = [self.namespaces] if namespace == Ellipsis else [namespace]
```
Comment (dannyroberts) [inline corehq/toggles.py]: I usually go for `is Ellipsis` but I'm not sure whether it makes a difference

**Review by dannyroberts — APPROVED** [review]

**Comment by dannyroberts** [comment]
Not sure if related:
```
======================================================================
FAIL: corehq.apps.data_interfaces.tests.tests:BulkArchiveForms.test_bulk_archive_basic
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/commcare-hq-ro/corehq/apps/data_interfaces/tests/tests.py", line 106, in test_bulk_archive_basic
    self.assertIn('We received your file and are processing it.', response.content)
AssertionError: 'We received your file and are processing it.' not found in '<!DOCTYPE html>\n\n<html lang="en">\n    <head>\n        \n        <title>\n            404 Not Found - CommCare HQ\n        </title>\n\n        <meta charset="utf-8">\n        <meta name="viewport" content="width=device-width, initial-scale=1">\n        <meta http-equiv="content-language" content="en">\n\n        \n        <link rel="shortcut icon" href="/static/style/images/favicon.png" />\n        <link type="application/opensearchdescription+xml"\n              rel="search"\n              href="/searchDescription.xml"\n              title="CommCare HQ" />\n\n        \n            <link type="text/less"\n                  rel="stylesheet"\n                  media="all"\n                  href="/static/style/less/bootstrap.debug.less" />\n            <link type="text/less"\n                  rel="stylesheet"\n
[...]
$(message_element).koApplyBindings({\n                alerts: message_alerts\n            });\n\n        </script>\n        \n    <script>\n        $(window).on(\'load\', function(){\n            $(\'#modalTrial30Day\').modal(\'show\');\n        });\n    </script>\n    \n\n    </body>\n</html>\n'
```

File: corehq/toggles.py [hunk 3]
```diff
@@ -70,29 +70,33 @@ def __init__(self, slug, label, tag, namespaces=None, help_link=None,
...
             return False
+
+        if namespace is not Ellipsis and namespace not in self.namespaces:
+            # short circuit if we're checking an item that isn't supported by this toggle
+            return False
+
         enabled_after = self.enabled_for_new_domains_after
         if (enabled_after is not None and NAMESPACE_DOMAIN in self.namespaces
-                and was_domain_created_after(item, enabled_after)):
+            and was_domain_created_after(item, enabled_after)):
```
Comment (dimagimon) [inline corehq/toggles.py]: E129 visually indented line with same indent as next logical line

**Pull request merged**